sdk/storage/Azure.Storage.Common/tests/Shared/TestExtensions.cs

```diff
@@ -65,6 +65,82 @@ public static Uri ToHttp(this Uri uri)
             return builder.Uri;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="StorageConnectionString"/> class using the specified
+        /// credentials, and specifies whether to use HTTP or HTTPS to connect to the storage services.
+        /// </summary>
+        /// <param name="storageCredentials">A StorageCredentials object.</param>
+        /// <param name="useHttps"><c>true</c> to use HTTPS to connect to storage service endpoints; otherwise, <c>false</c>.</param>
+        /// <remarks>Using HTTPS to connect to the storage services is recommended.</remarks>
+        internal static StorageConnectionString CreateStorageConnectionString(object storageCredentials, bool useHttps) =>
```

Comment:
As an FYI, this and the bodies of the other new factory methods were extracted as-is from StorageConnectionString constructors, which were only used by tests. 

The parameter is object, because it can be either a SharedAccessSignatureCredentials or a StorageSharedKeyCredential. We probably didn't have them inherit from a common type so as to avoid adding types that customers wouldn't actually need to use.

Reply:
Azure.Core didn't want to add a base `Credential` type so we're stuck with `object`.